### ISSUE TITLE
Compilation errors on Ubuntu 16.04

### ISSUE THREAD

**Issue by csverma610** [libigl/libigl#282]
Hello,

I had no problems installing igl on Ubuntu 14.04, but it seems that both the g++ compiler
and Eigen software is different on 16.04. I am having compiling the following files:

```
    orient_outward.cpp orth.cpp \
    is_symmetric.cpp  setdiff.cpp \
    min_quad_dense.cpp min_quad_with_fixed.cpp mode.cpp \
    integrable_polyvector_fields.cpp \
    nchoosek.cpp next_filename.cpp normal_derivative.cpp normalize_quat.cpp \
    point_simplex_squared_distance.cpp writeSTL.cpp \
    project_isometrically_to_plane.cpp project_to_line.cpp project_to_line_segment.cpp \
    cut_mesh_from_singularities.cpp cut_mesh.cpp dot_row.cpp dot.cpp eigs.cpp \
    infinite_cost_stopping_condition.cpp
```

For example:
/usr/include/eigen3/Eigen/src/Core/PlainObjectBase.h:457:25: error: ‘Eigen::PlainObjectBase<Derived>::PlainObjectBase() [with Derived = Eigen::Matrix<double, -1, -1>]’ is protected
     EIGEN_STRONG_INLINE PlainObjectBase() : m_storage()                        ^
orient_outward.cpp:43:29: error: within this context
   PlainObjectBase<DerivedV> N,BC,BCmean;

and

is_symmetric.cpp: In instantiation of ‘bool igl::is_symmetric(const Eigen::PlainObjectBase<Derived>&) [with DerivedA = Eigen::Matrix<double, -1, -1>]’:
is_symmetric.cpp:76:148:   required from here
/usr/include/eigen3/Eigen/src/Core/PlainObjectBase.h:505:25: error: ‘Eigen::PlainObjectBase<Derived>::PlainObjectBase(const Eigen::DenseBase<OtherDerived>&) [with OtherDerived = Eigen::Transpose<const Eigen::Matrix<double, -1, -1> >; Derived = Eigen::Matrix<double, -1, -1>]’ is protected
     EIGEN_STRONG_INLINE PlainObjectBase(const DenseBase<OtherDerived> &other)
                         ^
is_symmetric.cpp:38:69: error: within this context
   const typename Eigen::PlainObjectBase<DerivedA>& AT = A.transpose();

Can someone help resolve these errors ? 

Thanks.

**Comment by s-koch** [comment]
Hi,
I could not reproduce your problem on Ubuntu 16.04. What commands did you run for compilation?

**Comment by alecjacobson** [comment]
What version of Eigen are you trying to use?

**Comment by csverma610** [comment]
Hello,

It seems that Ubuntu 16.04 is shipping Eigen 3.3 beta. I have been trying
to create a static library of
libigl. Using the following command, I think, we should be able to
reproduce the errors.

CPPFLAGS = -O3 -std=c++11 -I/usr/include/eigen3  -DIGL_STATIC_LIBRARY

Thanks.
csv

**Comment by csverma610** [comment]
Hello,

I have been trying to create static library and the command I gave was:

CPPFLAGS = -O3 -std=c++11 -I/usr/include/eigen3  -DIGL_STATIC_LIBRARY

I hope you are able to reproduce the errors with this command.

csv

**Comment by alecjacobson** [comment]
Weird that it ships with a beta version. We should probably prepare libigl
to work with the latest greatest Eigen, but for now only the actual release
is supported.

**Comment by csverma610** [comment]
Hello Alec,

I installed Eigen 3.2.8 and it seems to compiling well. However, I get tons
to warning messages concerning
deprecated features which we do not get from new Eigen Beta version.

I fixed one error with Eigen3.2.8. Did you miss putting std:: in front of
isinf ?

infinite_cost_stopping_condition.cpp: In lambda function:
infinite_cost_stopping_condition.cpp:59:24: error: ‘isinf’ was not declared
in this scope
       return isinf(cost);


csv

**Comment by csverma610** [comment]
Hello,

It seems that many peoples have already raised the issues of Eigen <= 3.2.8
and C++ (>11) and as
I understand it has been resolved in Eigen beta version. One of the blogs
says:

The usage of the deprecated binder* classes has been removed in 3.3-beta1.

It gives maximum number of warnings.

csv

**Comment by alecjacobson** [comment]
oy vey.

Yes, this will imminently become an issue.

I think you're right about those inf related issues. If you pull request I
will merge.

**Comment by s-koch** [comment]
The missing std:: is already fixed with #281.

**Comment by alecjacobson** [comment]
As far as I can tell, _these_ issues with Eigen 3.3.* are fixed (though still not upgrading to Eigen 3.3.* due to outstanding bugs in their releases)
